src/Domains/Subscription/Plans/Repositories/PlanRepository.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kanvas\Subscription\Plans\Repositories;
+
+use Illuminate\Database\Eloquent\Model;
+use Kanvas\Subscription\Plans\Models\Plan;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Kanvas\Exceptions\ModelNotFoundException as ExceptionsModelNotFoundException;
+use Baka\Contracts\AppInterface;
+use Baka\Traits\SearchableTrait;
+
+class PlanRepository
+{
+    use SearchableTrait;
+
+    /**
+     * Get the model instance for Plan.
+     *
+     * @return Model
+     */
+    public static function getModel(): Model
+    {
+        return new Plan();
+    }
+
+    public static function getByIdWithApp(int $id, ?AppInterface $app = null): Model
```

Comment:
We have this functionality in the base model [Check this file](https://github.com/bakaphp/kanvas-ecosystem-api/blob/35b12b116491547392aa254340b111fccf593240/src/Baka/Traits/KanvasModelTrait.php#L65)